### DISCUSSION TITLE
GTSAM_USE_SYSTEM_EIGEN=On by default?

### DISCUSSION THREAD

**Discussion by jlblancoc** [borglab/gtsam#1422] [General]
Hi, 

Any reason `GTSAM_USE_SYSTEM_EIGEN` shouldn't be set to `On` by default in `develop`, at least targeting v4.3?

If you prefer `GTSAM_USE_SYSTEM_EIGEN` to still be `Off` **if** no system version is found first, I could do a PR for it, so the change does not break anyone's build...

PS: main reason to have `GTSAM_USE_SYSTEM_EIGEN=On` by default is to avoid crashes due to wrong memory alignment and/or layouts if mixing different Eigen versions in gtsam vs the user code, e.g. input or output matrices/vectors from other libraries that also use Eigen.

**Comment by dellaert** [comment]
So, maybe this is a debate: Eigen 3.4 actually gets rid of all alignment problems. If we want to *use* that functionality, then we should insist on that version, and so we can only use system if it has the right version. I think that is the version on the latest Ubuntu, so I would be fine with that if we check the version.

**Reply by ProfFan** [reply]
I think what @jlblancoc is referring to is NOT the fixed sized matrix alignment problem - it is the problem when you have multiple copies of different versions of Eigen. The fixed sized alignment problem should not matter that much - we have the Eigen alignment macros in all our data structures, which is available even before 3.4.

**Reply by dellaert** [reply]
Well, yes, but an advantage of Eigen 3.4 is that we can get rid off all that.

**Reply by jlblancoc** [reply]
Right: targeting C++17 allows ones to forget about the memory alignment issues. I was more concerned with potential memory layout differences between different versions of eigen3 (this has happened in the past...). 

I see Eigen 3.4 has been in Ubuntu repos since 22.04 (jammy), so, what about a PR that: 
- Detect eigen3 in the system.
- If system eigen3 >=3.4, use it via setting the *default* value of GTSAM_USE_SYSTEM_EIGEN to On.
- Otherwise (or if not found), keep using the embedded copy of 3.4.

Sounds ok? In that way we could avoid the installation of all eigen3 headers side by side with gtsam headers, effectively exposing eigen 3.4 twice (in different directories) to users.
